code/FulpstationCode/fulp_mobs/apostle.dm

```diff
@@ -0,0 +1,184 @@
+	deathmessage = "evaporates in a moment, leaving heavenly light and feathers behind."
+	deathsound = 'sound/tegu_sounds/mob/apostle_death.ogg'
+	attack_action_types = list(/datum/action/innate/megafauna_attack/holy_revival,
+							   /datum/action/innate/megafauna_attack/fire_field)
+	small_sprite_type = /datum/action/small_sprite/megafauna/tegu/angel
+	var/holy_revival_cooldown = 20 SECONDS
+	var/holy_revival_cooldown_base = 20 SECONDS
+	var/fire_field_cooldown = 30 SECONDS
+	var/fire_field_cooldown_base = 30 SECONDS
+	var/apostle_cooldown = 120 SECONDS //Cooldown for revival of non-apostles.
+	var/apostle_cooldown_base = 120 SECONDS
+	var/apostle_num = 1 //Number of apostle. Used for revival and finale.
+	var/apostle_line
+	var/apostle_prev //Used for previous apostle's name, to reference in next line.
+
+/mob/living/simple_animal/hostile/megafauna/apostle/Initialize()
+	. = ..()
+
+/mob/living/simple_animal/hostile/megafauna/apostle/ex_act(severity, target)
+	return //Resistant to explosions
+
+/datum/action/small_sprite/megafauna/tegu
+	small_icon = 'icons/Fulpicons/megafauna.dmi'
+
+/datum/action/small_sprite/megafauna/tegu/angel
+	small_icon_state = "angel_small"
+
+/datum/action/innate/megafauna_attack/holy_revival
+	name = "Holy Revival"
+	icon_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "magicm"
+	chosen_message = "<span class='colossus'>You are now reviving the dead, to join your cause.</span>"
+	chosen_attack_num = 1
+
+/datum/action/innate/megafauna_attack/fire_field
+	name = "Holy Revival"
+	icon_icon = 'icons/effects/fire.dmi'
+	button_icon_state = "fire"
+	chosen_message = "<span class='colossus'>You are now reviving the dead, to join your cause.</span>"
+	chosen_attack_num = 2
+
+/mob/living/simple_animal/hostile/megafauna/apostle/OpenFire()
+	if(client)
+		switch(chosen_attack)
+			if(1)
+				revive_humans()
+			if(2)
+				fire_field()
+		return
+
+/mob/living/simple_animal/hostile/megafauna/apostle/proc/revive_humans()
+	if(holy_revival_cooldown > world.time)
+		return
+	holy_revival_cooldown = (world.time + holy_revival_cooldown_base)
+	playsound(src, 'sound/effects/pray.ogg', 250, 1)
+	for(var/i in range(3, src))
+		if(isturf(i))
+			new /obj/effect/temp_visual/cult/sparks(i)
+			continue
+		if(ishuman(i))
+			var/mob/living/carbon/human/H = i
+			if(!("apostle" in H.faction))
+				if(H.stat == DEAD && apostle_cooldown <= world.time) // && H.mind)
+					apostle_cooldown = (world.time + apostle_cooldown_base)
+					H.set_species(/datum/species/human, 1)
+					H.regenerate_limbs()
+					H.regenerate_organs()
+					H.dna.species.GiveSpeciesFlight(H)
+					H.revive(full_heal = TRUE, admin_revive = FALSE)
+					H.grab_ghost(force = TRUE)
+					to_chat(H, "<span class='notice'>You feel protected by holy light!</span>")
+					// Giving the fancy stuff to new apostle
+					//H.mind.add_antag_datum(/datum/antagonist/apostle)
+					H.faction |= "apostle"
+					ADD_TRAIT(H, TRAIT_BOMBIMMUNE, SPECIES_TRAIT)
```

Comment:
So here they can give someone wings, then makes them immune to bombs, fire, lowpressure and cold, while also changing their faction?

Reply:
It is supposed to switch them to the side of sentient mob. So far it wouldn't work as normal megafauna.

Reply:
So if theyre lucky enough to have this happen on a Captain/HoS/Geared up miner, they can just 'bone the entire station?

Reply:
It's up to whoever is playing as the "megafauna". However, they are being given an absolutely new set of items when the number of apostles approaches 12.